sdk/tests/deqp/modules/shared/glsShaderExecUtil.js

```diff
@@ -446,7 +446,11 @@ goog.scope(function() {
             else if (gluShaderUtil.isDataTypeMatrix(basicType)) {
                 /** @type {number} */ var numRows = gluShaderUtil.getDataTypeMatrixNumRows(basicType);
                 /** @type {number} */ var numCols = gluShaderUtil.getDataTypeMatrixNumColumns(basicType);
-                /** @type {number} */ var stride = numRows * numCols * 4;//sizeof(float);
+                // A matrix consists of several (column-major) vectors. A buffer is created for
+                // every vector in gluDrawUtil.draw() below. Data in every buffer will be tightly
+                // packed. So the stride should be 0. This is different from the code in native
+                // deqp, which use only one buffer for a matrix, the data is interleaved.
+                /** @type {number} */ var stride = 0;
```

Comment:
Thanks for your feedback @Richard-Yunchao regarding how much work it would be to re-port this code. I think we will stick with its current structure for now. It was a great effort to port it over in the first place and I think it is most expedient to diagnose and fix the bugs in the current code.

I don't think it is feasible to write a tool to autogenerate the JavaScript from C++. The working group rejected the idea of using Emscripten to get the native dEQP tests running; the code is unreadable, and doesn't integrate well with surrounding JavaScript code. A custom tool to port code would be a lot of work and it would be unclear whether it would actually function as expected. This is why we contracted Mobica to do the dEQP port by hand in the first place.